lib/controller.js

```diff
@@ -452,6 +455,165 @@ controller.connectToNetwork = function(opts) {
+        }
+
+        // If the update is forced or this build was requested
+        if (opts.build) {
+          // Fetch the requested build
+          return controller.updateTesselWithVersion(tessel, currentVersion, requestedVersion)
+            .then(resolve, reject);
+        } else {
+          // If they have requested the latest firmware
+          var verifiedVersion;
+          if (requestedVersion === 'latest') {
+            // Figure out what the version is
+            verifiedVersion = updates.fetchBuildList()
+              .then(function(builds) {
+                // Make sure we actually have builds
+                if (Object.keys(builds).length === 0) {
+                  return reject('Could not detect any builds on server.');
+                }
+                // Sort the builds by version
+                var buildArray = updates.buildDictToSortedArray(builds);
+
+                // Return the newest one
+                return buildArray[0].version;
+              });
+          }
+          // If they requested a specific version
+          else {
+            // Ensure that it exists by fetching the builds
+            verifiedVersion = updates.fetchSHAForBuildVersion(requestedVersion)
+              // If that function resolves, the sha exists so the version exists
+              .then(function() {
+                return requestedVersion;
+              })
+              // The version does not exist
+              .catch(reject);
+          }
+          // Once we have our verified version
+          verifiedVersion.then(function(requestedVersion) {
+            // Check if the current build is the same or newer if this isn't a forced update
+            if (!opts.force && currentVersion >= requestedVersion) {
+              // If it's not, close the Tessel connection and print the error message
+              var errMessage = tessel.name + ' is already on the latest firmware build. You can force an update with “tessel update --force”.';
+              return controller.closeTesselConnections([tessel])
+                .then(reject.bind(this, errMessage), reject.bind(this, errMessage));
+            } else {
+              if (!opts.force) {
+                // If it is a newer version, let's update...
+                logs.info('New firmware version found...' + requestedVersion);
+              }
+
+              logs.info('Updating ' + tessel.name + ' to latest build...');
+              // Fetch the requested build
+              return controller.updateTesselWithVersion(tessel, currentVersion, requestedVersion)
+                .then(resolve, reject);
+            }
+          });
+        }
+      });
+    }
+  });
+};
+
+controller.updateTesselWithVersion = function(tessel, currentVersion, requestedVersion) {
+  // Fetch the requested build
+  return updates.fetchBuild(requestedVersion)
+    .then(function startUpdate(build) {
```

Comment:
@Frijol the whole process takes ~3 minutes right now. Maybe a bit longer depending on connection speeds. We should definitely have blinky lights eventually. (Will make an issue for it).